apps/air-discount-scheme/backend/src/app/modules/user/user.controller.ts

```diff
@@ -79,13 +79,20 @@ export class PrivateUserController {
         '[/relations] Request parameters do not correspond with user authentication.',
       )
     }
-    const relations: string[] = await this.userService.getRelations(authUser)
-    const users = (await Promise.all([
-      this.userService.getUserInfoByNationalId(authUser.nationalId),
-      ...relations.map((nationalId) =>
-        this.userService.getUserInfoByNationalId(nationalId),
-      ),
-    ])) as User[]
-    return users.filter((user) => user) as User[]
+    const relations = await this.userService.getRelations(authUser)
+
+    // Adding User in beginning of array so order is correct.
+    relations.unshift(authUser.nationalId)
+
+    const userAndRelatives = await this.userService.getMultipleUsersByNationalIdArray(
+      relations,
+    )
+
+    if (Array.isArray(userAndRelatives) && userAndRelatives[0] === null) {
+      throw new Error(
+        'Could not find NationalRegistry records of both User and relatives.',
+      )
+    }
+    return userAndRelatives as Array<User>
```

Comment:
"this is a lie" since there might be nulls in there.

Should either filter them out here, or update the schema and return type to match what is actually returned.

Reply:
fixed :+1: